### PR TITLE
[CLANG][DRIVER] Resolve Argument Visibility and Duplication Issue

### PR DESCRIPTION
**Pull request by jle-quel**
# Description

This PR is a response to the issue (https://github.com/intel/llvm/issues/11624) which fixes the `-Xcuda-ptxas` flag not being recognised by the driver in cl-mode; Changing the visibility to that option fixed that problem.

# Modifications

Furthermore, it has been noticed that the arguments are being passed twice to `ptxas`.
This also has been fixed by filtering out the arguments before appending them to the new `DAL` created by `CudaToolChain::TranslateArgs`.

### CODE REVIEW

**Comment by bader** [comment]
@jle-quel, please, add a regression test. 
If this patch doesn't depend on other patches which are not upstreamed yet, it should go to llvm-project repository.

File: clang/lib/Driver/ToolChains/Cuda.cpp [hunk 1]
```diff
@@ -1151,7 +1151,9 @@ CudaToolChain::TranslateArgs(const llvm::opt::DerivedArgList &Args,
   }
 
   for (Arg *A : Args) {
-    DAL->append(A);
```
Comment (srividya-sundaram) [inline clang/lib/Driver/ToolChains/Cuda.cpp:1154]: nit: Adding a comment here could be helpful.
Please add a test to check if `-Xcuda-ptxas` is recognized in `clang-cl` mode as well as a test to check if the arguments are passed to the `PTXAS` just once.

**Review by mdtoguchi — APPROVED** [review]

**Comment by jle-quel** [comment]
Thank you for the reviews, I will close this PR for an upstream llvm PR.

**Comment by jle-quel** [comment]
Upstream PR: https://github.com/llvm/llvm-project/pull/86807

**Comment by jle-quel** [comment]
Merged Upstream :ok:

**Pull request closed without merging**